resources/js/components/Tool.vue

```diff
@@ -1,27 +1,308 @@
+            <sortable-icon
+                @sort="sortByChange"
+                @reset="resetOrderBy"
+                resource-name="Pages"
+                uri-key="ga:entrances"
+                :direction="direction"
+            >
+              <span class="inline-flex items-center">
+                {{ __('Entrances') }}
+              </span>
+            </sortable-icon>
+          </th>
+          <th class="text-left">
+            <sortable-icon
+                @sort="sortByChange"
+                @reset="resetOrderBy"
+                resource-name="Pages"
+                uri-key="ga:bounceRate"
+                :direction="direction"
+            >
+              <span class="inline-flex items-center">
+                {{ __('Bounce Rate') }}
+              </span>
+            </sortable-icon>
+          </th>
+          <th class="text-left">
+            <sortable-icon
+                @sort="sortByChange"
+                @reset="resetOrderBy"
+                resource-name="Pages"
+                uri-key="ga:exitRate"
+                :direction="direction"
+            >
+              <span class="inline-flex items-center">
+                {{ __('Exit Rate') }}
+              </span>
+            </sortable-icon>
+          </th>
+          <th class="text-left">
+            <sortable-icon
+                @sort="sortByChange"
+                @reset="resetOrderBy"
+                resource-name="Pages"
+                uri-key="ga:pageValue"
+                :direction="direction"
+            >
+              <span class="inline-flex items-center">
+                {{ __('Page Value') }}
+              </span>
+            </sortable-icon>
+          </th>
+          </thead>
+          <tbody>
+          <tr v-for="page in pages">
+            <td>{{ page.name }}</td>
+            <td>{{ page.path }}</td>
+            <td>{{ page.visits }}</td>
+            <td>{{ page.unique_visits }}</td>
+            <td>{{ getFormattedTime(page.avg_page_time) }}</td>
+            <td>{{ page.entrances }}</td>
+            <td>{{ getFormattedPercent(page.bounce_rate) }}</td>
+            <td>{{ getFormattedPercent(page.exit_rate) }}</td>
+            <td>{{ getFormattedCurrency(page.page_value) }}</td>
+          </tr>
+          </tbody>
+        </table>
+      </div>
 
-            <p class="text-white-50% text-lg">
-                You can edit this tool's component at:
-                <code class="ml-1 border border-80 text-sm font-mono text-white bg-black rounded px-2 py-1">/nova-tools/{{ class }}/resources/js/components/Tool.vue</code>
-            </p>
-        </card>
-    </div>
+      <pagination-links
+          :data="pages"
+          :hasMore="hasMore"
+          :hasPrevious="hasPrevious"
+          :current-page="page"
+          :total-pages="totalPages"
+          @previous="previousPage"
+          @next="nextPage"
+      ></pagination-links>
+    </loading-card>
+  </loading-view>
 </template>
 
 <script>
```

Comment:
I find myself again wanting to have the ability to figure out how much of this code is our custom code and how much is just reproducing existing Nova functionality.

What do you think of the idea of, after merging this PR, we band together and pair some time and try our hands at extracting a plugin or something like it that would be basically "here's how to make a Nova tool that looks like the resource pages but is using a non-Eloquent source of data"?

Reply:
👍  Sounds Good!